django/forms/widgets.py

```diff
@@ -268,6 +268,7 @@ class Input(Widget):
 
     def __init__(self, attrs=None):
         if attrs is not None:
+            attrs = attrs.copy()
             self.input_type = attrs.pop('type', self.input_type)
```

Comment:
Sorry - I understand that. My follow up comment above also highlights another potential issue that your change solves anyway.